### PR TITLE
Remove unused `type: ignore`

### PR DESCRIPTION
**Pull request by johnnv1**
- Solve some previous type annotations errors (related to #1894)
- Add `warn_unused_ignores = True` to mypy config (related to #1992)
  - Fix all unused `type: ignore` pointed out by mypy
- Add `kornia.core` for these files

### CODE REVIEW

File: kornia/nerf/data_utils.py [hunk 1]
```diff
@@ -12,6 +13,14 @@
 RayGroup = Tuple[Tensor, Tensor, Optional[Tensor]]
 
 
+def _is_list_of_str(lst: Sequence[object]) -> TypeGuard[List[str]]:
```
Comment (johnnv1) [inline kornia/nerf/data_utils.py:16]: Maybe these functions should be in a generatl `utils`?

**Comment by johnnv1** [comment]
The errors at CI-typing: 
```out
kornia/io/io.py 
 - 37: error: Module has no attribute "dlpack"  [attr-defined]

kornia/testing/__init__.py
 - 286: error: Incompatible import of "assert_close"
 - 260: error: Module "torch.testing" has no attribute
 ```
They are due to my changes, but I need to reproduce them locally to correct them. The rest are errors that already existed.

**Comment by edgarriba** [comment]
> kornia/io/io.py 
>  - 37: error: Module has no attribute "dlpack"  [attr-defined]
i noticed that some pytorch versions have not well defined dlpack packaging. Worth to open an issue in upstream pytorch ?

> kornia/testing/__init__.py
>  - 286: error: Incompatible import of "assert_close"
>  - 260: error: Module "torch.testing" has no attribute

related to this: https://github.com/kornia/kornia/blob/master/kornia/testing/__init__.py#L290

as per discussed in slack, i advocated to starting dropping support from the tail e.g 1.9 which seems a more stable api version.
@johnnv1 are you in our slack btw ? xd

**Comment by johnnv1** [comment]
I'm at slack, but I hadn't seen this thread, I will look into it now

**Comment by edgarriba** [comment]
@johnnv1 Should we merge this for now ?

File: kornia/nerf/data_utils.py (hunk 1 above)
Comment (edgarriba) [reply kornia/nerf/data_utils.py:16]: Or as KORNIA_CHECK_

File: kornia/testing/__init__.py [hunk 2]
```diff
@@ -255,41 +255,48 @@ def _default_tolerances(*inputs: Any) -> Tuple[float, float]:
     return max(rtols), max(atols)
 
 
-try:
-    # torch.testing.assert_close is only available for torch>=1.9
-    from torch.testing import assert_close as _assert_close  # type: ignore
-
```
Comment (edgarriba) [inline kornia/testing/__init__.py:261]: cc @pmeier

**Review by edgarriba — APPROVED** [review]

**Comment by johnnv1** [comment]
For me, can ship it. but if you find it interesting to move those checkers to KORNIA_CHECK*, I can do it too

**Comment by edgarriba** [comment]
> For me, can ship it. but if you find it interesting to move those checkers to KORNIA_CHECK*, I can do it too

sure

**Pull request merged**